# Homework Assignment - Lesson 1: Introduction to R

**Student Name:** Francisco Guadarrama  
**Date:** 9/10/2025  
**Course:** Data Management  

---

## Instructions

Complete all the tasks below by adding your R code in the code cells and your written responses in markdown cells. Make sure to run all cells to show your output.

**📋 SUBMISSION**: When you're done, see [GITHUB_CLASSROOM_SUBMISSION.md](../../GITHUB_CLASSROOM_SUBMISSION.md) for complete submission instructions.

---

## Part 1: Setting Up Your Environment and Importing Data

### 1.1 Working Directory

Check your current working directory using `getwd()`. You should be in the main repository folder.

In [1]:
# Check current working directory
getwd()

[1] "/workspaces/assignment-1-version-3-zfrank33/assignment/Homework"

I was using the wrong path to the sales_data.csv and I mistakenly thought I was in the wrong directory so I tried changing before realizing my mistake and copied as path.

In [3]:
setwd('/workspaces/Data-Management-Assignment-1-Intro-to-R')

ERROR: Error in setwd("/workspaces/Data-Management-Assignment-1-Intro-to-R"): cannot change working directory


### 1.2 Package Loading

Load the required packages: `tidyverse` and `readxl`

In [2]:
# Load required packages
library(tidyverse)
library(readxl)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.2     ✔ tibble    3.3.0
✔ lubridate 1.9.4     ✔ tidyr     1.3.1
✔ purrr     1.1.0     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


### 1.3 Data Import - CSV

Import the `sales_data.csv` file from the `data/` folder into a data frame named `sales_df`.

In [3]:
# Import sales data
sales_df <- read_csv("/workspaces/assignment-1-version-3-zfrank33/data/sales_data.csv")
print("Sales data imported successfully!")

Rows: 100 Columns: 6
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr  (2): Product, Region
dbl  (3): TransactionID, Amount, Quantity
date (1): Date

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


[1] "Sales data imported successfully!"


I used list.files when I thought I was in the wrong directory before I realized I was using the wrong path.


In [4]:
getwd()  # Shows current directory
list.files()  # Shows files in current directory

[1] "/workspaces/assignment-1-version-3-zfrank33/assignment/Homework"

[1] "homework_lesson_1.ipynb" "README.md"

### 1.4 Data Import - Excel

Import both sheets from the `customer_feedback.xlsx` file.

In [5]:
# Import ratings data from Excel
ratings_df <- read_excel("/workspaces/assignment-1-version-3-zfrank33/data/customer_feedback.xlsx", sheet = "ratings")

# Import comments data from Excel  
comments_df <- read_excel("/workspaces/assignment-1-version-3-zfrank33/data/customer_feedback.xlsx", sheet = "customer_feedback")

print("Excel data imported successfully!")

[1] "Excel data imported successfully!"


---

## Part 2: Basic Data Inspection

For each of the three data frames you imported, perform the following inspection tasks. Add both your R code and a brief interpretation of what you observe.

### 2.1 Sales Data Inspection

In [6]:
# First 10 rows of sales data
print("First 10 rows of sales_df:")
head(sales_df, 10)

[1] "First 10 rows of sales_df:"


TransactionID Date       Product  Amount  Quantity Region
1   1            2023-01-27 Keyboard 1039.05 1        South 
2   2            2023-05-18 Tablet    597.55 5        East  
3   3            2023-02-14 Keyboard 1539.48 1        West  
4   4            2023-07-28 Mouse     261.66 1        South 
5   5            2023-05-17 Monitor  1047.73 1        South 
6   6            2023-02-25 Keyboard 1824.05 2        East  
7   7            2023-01-20 Monitor   475.83 1        North 
8   8            2023-07-14 Mouse     758.05 5        North 
9   9            2023-10-02 Mouse    1717.20 4        South 
10 10            2023-10-15 Laptop   1437.21 3        North

There are sales data for the year 2023 (at least in the first ten rows) across 4 regions. The sales data contains 6 variables, TransID, date, product, amount, quantity, and region.


In [7]:
# Structure of sales data
print("Structure of sales_df:")
str(sales_df)

[1] "Structure of sales_df:"
spc_tbl_ [100 × 6] (S3: spec_tbl_df/tbl_df/tbl/data.frame)
 $ TransactionID: num [1:100] 1 2 3 4 5 6 7 8 9 10 ...
 $ Date         : Date[1:100], format: "2023-01-27" "2023-05-18" ...
 $ Product      : chr [1:100] "Keyboard" "Tablet" "Keyboard" "Mouse" ...
 $ Amount       : num [1:100] 1039 598 1539 262 1048 ...
 $ Quantity     : num [1:100] 1 5 1 1 1 2 1 5 4 3 ...
 $ Region       : chr [1:100] "South" "East" "West" "South" ...
 - attr(*, "spec")=
  .. cols(
  ..   TransactionID = col_double(),
  ..   Date = col_date(format = ""),
  ..   Product = col_character(),
  ..   Amount = col_double(),
  ..   Quantity = col_double(),
  ..   Region = col_character()
  .. )
 - attr(*, "problems")=<externalptr> 


The file contains 100 rows (observations/transactions) × 6 columns (variables) and it's a tibble. The transID is a float and a unique identifier. Date is a date variable, Rroduct is a string, Amount is formatted as currency, and Region is string.

In [8]:
# Summary statistics for sales data
print("Summary statistics for sales_df:")
summary(sales_df)

[1] "Summary statistics for sales_df:"


 TransactionID         Date              Product              Amount       
 Min.   :  1.00   Min.   :2023-01-01   Length:100         Min.   :  58.06  
 1st Qu.: 25.75   1st Qu.:2023-03-20   Class :character   1st Qu.: 520.79  
 Median : 50.50   Median :2023-07-03   Mode  :character   Median : 960.75  
 Mean   : 50.50   Mean   :2023-06-27                      Mean   : 962.71  
 3rd Qu.: 75.25   3rd Qu.:2023-09-25                      3rd Qu.:1384.63  
 Max.   :100.00   Max.   :2023-12-26                      Max.   :1985.60  
    Quantity       Region         
 Min.   :1.00   Length:100        
 1st Qu.:1.00   Class :character  
 Median :3.00   Mode  :character  
 Mean   :2.69                     
 3rd Qu.:4.00                     
 Max.   :5.00                     

**Your observations about sales_df:**

There are sales data for the year 2023 (at least in the first ten rows) across 4 regions. The sales data contains 6 variables, TransID, date, product, amount, quantity, and region. The file contains 100 rows (observations/transactions) × 6 columns (variables) and it's a tibble. The transID is a float and a unique identifier. Date is a date variable, Rroduct is a string, Amount is formatted as currency, and Region is string. I don't see why transaction id is a float and doesn't just increment by 1.

### 2.2 Ratings Data Inspection

In [9]:
# TODO: Add your code here for ratings_df inspection
# - First 10 rows
head(ratings_df)
# - Structure  
str(ratings_df)
# - Summary statistics
summary(ratings_df)

CustomerID ProductRating ServiceRating OverallSatisfaction
1 1          1             4             3                  
2 2          3             2             4                  
3 3          1             5             5                  
4 4          1             2             3                  
5 5          5             1             2                  
6 6          4             2             2                  
  RecommendationScore
1 2                  
2 7                  
3 1                  
4 6                  
5 2                  
6 6

tibble [50 × 5] (S3: tbl_df/tbl/data.frame)
 $ CustomerID         : num [1:50] 1 2 3 4 5 6 7 8 9 10 ...
 $ ProductRating      : num [1:50] 1 3 1 1 5 4 3 5 4 4 ...
 $ ServiceRating      : num [1:50] 4 2 5 2 1 2 5 2 1 1 ...
 $ OverallSatisfaction: num [1:50] 3 4 5 3 2 2 5 5 3 2 ...
 $ RecommendationScore: num [1:50] 2 7 1 6 2 6 5 10 6 0 ...


   CustomerID    ProductRating ServiceRating  OverallSatisfaction
 Min.   : 1.00   Min.   :1.0   Min.   :1.00   Min.   :1.00       
 1st Qu.:13.25   1st Qu.:2.0   1st Qu.:1.00   1st Qu.:2.00       
 Median :25.50   Median :3.0   Median :3.00   Median :3.00       
 Mean   :25.50   Mean   :3.2   Mean   :2.94   Mean   :2.92       
 3rd Qu.:37.75   3rd Qu.:5.0   3rd Qu.:5.00   3rd Qu.:4.00       
 Max.   :50.00   Max.   :5.0   Max.   :5.00   Max.   :5.00       
 RecommendationScore
 Min.   : 0.0       
 1st Qu.: 1.0       
 Median : 4.5       
 Mean   : 4.2       
 3rd Qu.: 7.0       
 Max.   :10.0       

**Your observations about ratings_df:**

There 50 customers (rows) and5 variables (columns) of rating data.

CustomerID is a unique identifier for each customer (1 through 50)

ProductRating is Product satisfaction ratings on a 1-5 scale with the Mean: 3.2 which is slightly above average

ServiceRating is service quality ratings on a 1-5 scale with the Mean: 2.94 which is slightly below average

OverallSatisfaction is overall experience ratings on a 1-5 scale with the Mean: 2.92 which is slightly below average

RecommendationScore is Likelihood to recommend on a scale of 0-10 and the Mean is 4.2 which is neutral (5 would be neutral)

Key Insights:
Product ratings (3.2) are higher than service ratings (2.94). Overall satisfaction (2.92) aligns more with service than product ratings which means that service quality appears to be dragging down overall satisfaction and recommendation scores. May want to investigate why service ratings are lower than product ratings and work to improve service quality to boost overall satisfaction and recommendation rates.

In [10]:
# TODO: Add your code here for comments_df inspection
# - First 10 rows
head(comments_df)
# - Structure
str(comments_df)
# - Summary statistics
summary(comments_df)

FeedbackID CustomerID Feedback_Text                    Contact_Info          
1 1          12         Highly recommend this item       bob.wilson@test.org   
2 2          40         Excellent service                555-123-4567          
3 3          34         Poor quality control             jane_smith@company.com
4 4           1         average product, nothing special jane_smith@company.com
5 5          47         AMAZING customer support!!!      555-123-4567          
6 6          13         AMAZING customer support!!!      john.doe@email.com    
  Feedback_Date
1 2024-02-23   
2 2024-01-21   
3 2023-09-02   
4 2023-08-21   
5 2023-04-24   
6 2023-04-16

tibble [100 × 5] (S3: tbl_df/tbl/data.frame)
 $ FeedbackID   : num [1:100] 1 2 3 4 5 6 7 8 9 10 ...
 $ CustomerID   : num [1:100] 12 40 34 1 47 13 13 37 49 23 ...
 $ Feedback_Text: chr [1:100] "Highly recommend this item" "Excellent service" "Poor quality control" "average product, nothing special" ...
 $ Contact_Info : chr [1:100] "bob.wilson@test.org" "555-123-4567" "jane_smith@company.com" "jane_smith@company.com" ...
 $ Feedback_Date: POSIXct[1:100], format: "2024-02-23" "2024-01-21" ...


   FeedbackID       CustomerID    Feedback_Text      Contact_Info      
 Min.   :  1.00   Min.   : 1.00   Length:100         Length:100        
 1st Qu.: 25.75   1st Qu.:15.00   Class :character   Class :character  
 Median : 50.50   Median :26.00   Mode  :character   Mode  :character  
 Mean   : 50.50   Mean   :26.27                                        
 3rd Qu.: 75.25   3rd Qu.:38.00                                        
 Max.   :100.00   Max.   :50.00                                        
 Feedback_Date                
 Min.   :2023-01-03 00:00:00  
 1st Qu.:2023-04-23 18:00:00  
 Median :2023-08-24 12:00:00  
 Mean   :2023-08-22 07:26:24  
 3rd Qu.:2023-12-07 06:00:00  
 Max.   :2024-03-30 00:00:00  

**Your observations about comments_df:**

FeedbackID: Unique identifier for each feedback entry (1 through 100)

CustomerID: Customer identifiers (range: 1-50)

Feedback_Text: chr [1:100] 

Contact_Info: chr [1:100] Mixed contact information: emails and phone numbers

Feedback_Date: POSIXct[1:100] Timestamps of when feedback was provided

**Reflection Question 1:** What did you observe about the structure and content of the data files you loaded? What insights can you share about the quality and organization of the data?

**Your Answer:**

There are no missing values shown in structure output and the data types for each variable are appropriate. The value ranges (Amount: $261-$1824, Quantity: 1-5 units) are reasonable and there is consistent formatting across records.

### Question 1: Data Types Analysis

Based on your inspection of `sales_df`, what are the data types of the `Date` and `Amount` columns? Are these data types appropriate for typical business analytics tasks involving sales data? Explain why or why not.

**Your Answer:**

Date Column: Data Type: Date col_date(format = "")
The data type is ideal because it allows for time series analysis, date operations like calculating days between events, and filtering by date ranges. It also allows seasonality analysis which can identify patterns by month, quarter, or seasons.

Amount Column: Data Type: Numeric/Double col_double(). Decimal numbers for currency amounts.
The data type is ideal because it allows mathematical operations, financial analysis and statistical analysis.


### Question 2: Data Quality Assessment

Looking at all three datasets, what potential data quality issues do you notice? Consider missing values, data types, and any unusual patterns.

**Your Answer:**

The contact info field combines email addresses and phone numbers in a single column, which makes it difficult to categorize or analyze contact methods. I recommend splitting them into separate columns, one for email and one for phone numbers.
The CustomerID field shows inconsistencies across datasets—ratings_df contains only 50 unique CustomerIDs while feedback_df appears to have multiple entries per customer. I would suggest verifying if all CustomerID align across datasets.

### Question 3: Analysis Readiness

Which of the three datasets appears most ready for analysis, and which would require the most preprocessing? Justify your answer.

**Your Answer:**

sales_df is the most ready for analysis because it employs perfect data types: Date as Date, numbers as numeric, text as character, no missing values (complete records with no null values). It has well-organized columns with logical names, appropriate value ranges (reasonable amounts and quantities).
feedback_df would require the most preprocessing due to having to split the contact info into two separate columns, one for email and one for phone number.

---

## Submission Checklist

Before submitting, make sure you have:

- [ ] Completed all code cells in Parts 1 and 2
- [ ] Run all cells to show output
- [ ] Added your observations for each dataset
- [ ] Answered all reflection questions in Part 3
- [ ] Added your name and date at the top
- [ ] Saved the notebook
- [ ] Committed and pushed your changes to GitHub

**Great work completing your first R data analysis assignment! 🎉**

---

## 🚀 Ready to Submit?

### Easy Submission Steps (No Command Line Required!):

1. **Save this notebook** (Ctrl+S or File → Save)

2. **Use VS Code Source Control**:
   - Click the **Source Control** icon in the left sidebar (tree branch symbol)
   - Click the **"+"** button next to your notebook file
   - Type a message: `Submit homework 1 - [Your Name]`
   - Click **"Commit"** 
   - Click **"Sync Changes"** or **"Push"**

3. **Verify on GitHub**: Go to your repository online and confirm your notebook appears with your completed work

**📖 Need help?** See [GITHUB_CLASSROOM_SUBMISSION.md](../../GITHUB_CLASSROOM_SUBMISSION.md) for detailed instructions.

**🎉 Congratulations on completing your first R data analysis assignment!**